dt-core/dt-storage.php

```diff
@@ -211,12 +211,20 @@ public static function upload_file( string $key_prefix = '', array $upload = [],
             }
 
             return [
-            'uploaded_key' => $key,
-            'uploaded_thumbnail_key' => $uploaded_thumbnail_key,
-            'uploaded_large_thumbnail_key' => $uploaded_large_thumbnail_key,
+                'uploaded' => true,
+                'uploaded_key' => $key,
+                'uploaded_thumbnail_key' => $uploaded_thumbnail_key,
+                'uploaded_large_thumbnail_key' => $uploaded_large_thumbnail_key,
+                'uploaded_msg' => null
             ];
         } catch ( Throwable $e ) {
-            return false;
+            return [
```

Comment:
@kodinkat. Since an error happened lets return an new error
`return new WP_Error`

In Admin UIs we can display the more details of the error message.
For user facing UI lets keep the error simpler. 
Something went wrong. Please try again or have an admin check the connections setting.